### PR TITLE
[CapturePromotion] dealloc_box doesn't mutate.

### PR DESCRIPTION
**Pull request by nate-chandler**
The instruction only deallocates the box, it doesn't destroy its contents.  It's even less mutating than a `destroy_value`, which is already regarded as non-mutating.

rdar://139235335

### CODE REVIEW

**Comment by nate-chandler** [comment]
@swift-ci please test

**Comment by nate-chandler** [comment]
@swift-ci please test source compatibility

**Review by meg-gupta — APPROVED** [review]
LGTM!

File: lib/SILOptimizer/Mandatory/CapturePromotion.cpp [hunk 1]
```diff
@@ -1071,15 +1071,10 @@ class NonEscapingUserVisitor
   ALWAYS_NON_ESCAPING_INST(StrongRelease)
   ALWAYS_NON_ESCAPING_INST(DestroyValue)
   ALWAYS_NON_ESCAPING_INST(EndBorrow)
+  ALWAYS_NON_ESCAPING_INST(DeallocBox)
+  ALWAYS_NON_ESCAPING_INST(EndAccess)
```
Comment (meg-gupta) [inline lib/SILOptimizer/Mandatory/CapturePromotion.cpp:1075]: Not related to this PR - I noticed `copy_value` is not handler here.

**Pull request merged**